### DISCUSSION TITLE
Consolidated dependency update PRs?

### DISCUSSION THREAD

**Discussion by jplock** [dropwizard/dropwizard#9228] [Development]
I know with Dependabot, you can [group](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#groups) dependency updates across multiple dependencies to a single PR. Can we do something similar with Renovate to have a single dependency PR per branch we are maintaining? Or, even better, a way to pin specific major versions we are tracking per branch and let the other Maven-dependencies (for example) be updated across all branches.

I'm primarily asking to cut down on the amount of notifications from both the `dropwizard` and `metrics` repositories.

**Comment by joschi** [comment]
> Can we do something similar with Renovate to have a single dependency PR per branch we are maintaining?

Renovate supports the same thing (with the possibility to either use existing presets or fine-tune the handling of multiple dependencies): https://docs.renovatebot.com/presets-group/

The question is whether we'd like to have this for all dependencies or for only for a specific subset.

Any thoughts?

**Reply by jplock** [reply]
Maybe start with grouping all the maven-related ones together since they seem to be upgraded more frequently than the core dependencies?

**Reply by joschi** [reply]
That's what I did in https://github.com/dropwizard/dropwizard/pull/9238.

See https://github.com/dropwizard/dropwizard/pull/9240 for an example PR created with the new configuration.

**Reply by jplock** [reply]
Nice! That works. What if we did one across all the dependencies or do you think that would become unmanageable?

**Reply by joschi** [reply]
I don't think this would be a substantial improvement. Instead of 10 individual PRs, you'd get a single, permanently updated PR with 10 changes until it's going to be merged.

If you want to reduce the number of notifications of , maybe customizing the watch settings would make sense? Issues, discussions, security alerts sounds like a useful combination if you want to largely ignore PRs.

<img width="440" alt="Event subscription dialog" src="https://github.com/user-attachments/assets/e4543037-0000-4946-886e-f8c8c9aed613">

**Reply by jplock** [reply]
Yea thanks. I’m going to adjust those. I wish there was a way to ignore PRs with specific tags. I’m still interested in new functionality and bug fixes but all the dependency ones are noisy. Thank you for the suggestions
